apps/src/templates/AppView.jsx

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var ProtectedStatefulDiv = require('./ProtectedStatefulDiv.jsx');
+
+/**
+ * Top-level React wrapper for our standard blockly apps.
+ */
+var AppView = React.createClass({
+  propTypes: {
+    renderCodeApp: React.PropTypes.func.isRequired,
+    onMount: React.PropTypes.func.isRequired
+  },
+
+  componentDidMount: function () {
+    this.refs.codeApp.getDOMNode().innerHTML = this.props.renderCodeApp();
+    this.props.onMount();
+  },
+
+  render: function () {
+    return <ProtectedStatefulDiv ref="codeApp" />;
+  }
```

Comment:
Brad, your understanding of the scenario I was describing is correct. I also think you are right that the problem of forgetting to unmount the div is unlikely to go unnoticed. The related problem of accidentally re-rendering the div is much more likely to go unnoticed and cause weird bugs. `ProtectedStatefulDiv` sounds like the way to go.